lib/kubernetes-deploy/runner.rb

```diff
@@ -123,15 +113,18 @@ def run
       predeploy_priority_resources(resources)
 
       phase_heading("Deploying all resources")
-      if PROTECTED_NAMESPACES.include?(@namespace) && @prune
+      if PROTECTED_NAMESPACES.include?(@namespace) && prune
         raise FatalDeploymentError, "Refusing to deploy to protected namespace '#{@namespace}' with pruning enabled"
       end
 
-      deploy_resources(resources, prune: @prune)
+      deploy_resources(resources, prune: prune)
 
-      return unless wait_for_completion?
+      return if skip_wait
       wait_for_completion(resources)
       report_final_status(resources)
+    rescue FatalDeploymentError => error
+      @logger.fatal "#{error.class}: #{error.message}"
+      raise error
```

Comment:
`raise` without argument from inside a rescue block re-raises the original exception. It's up to you but `raise` and `raise error` would have the same result here.